clients/ts/signalr/src/HubConnection.ts

```diff
@@ -31,26 +26,21 @@ export class HubConnection {
     private id: number;
     private closedCallbacks: Array<(error?: Error) => void>;
     private timeoutHandle: NodeJS.Timer;
-    private timeoutInMilliseconds: number;
     private receivedHandshakeResponse: boolean;
 
-    constructor(url: string, options?: IHubConnectionOptions);
-    constructor(connection: IConnection, options?: IHubConnectionOptions);
-    constructor(urlOrConnection: string | IConnection, options: IHubConnectionOptions = {}) {
-        options = options || {};
+    public serverTimeoutInMilliseconds: number;
 
-        this.timeoutInMilliseconds = options.timeoutInMilliseconds || DEFAULT_TIMEOUT_IN_MS;
+    constructor(connection: IConnection, logger: ILogger, protocol: IHubProtocol) {
```

Comment:
So clean 😄  I'm just sad it isn't a connection factory 😢

Reply:
Might still be able to make that happen ;)